app/src/main/java/com/gophillygo/app/activities/BaseAttractionActivity.java

```diff
@@ -168,8 +176,10 @@ public void locationFound(Location location) {
         Log.d(LOG_LABEL, "location found: " + location.toString());
         currentLocation = location;
         locationHasChanged = true;
-        nearestDestinations = findNearestDestinations();
-        locationOrDestinationsChanged();
+        if (destinationInfos != null) {
```

Comment:
Why is it not okay to clear the list of destinations when none are returned? Why does the change handler not get fired in that case?